BigQuery/tests/System/LoadDataAndQueryTest.php

```diff
@@ -821,4 +836,33 @@ public function testLoadAndQueryDataWithHivePartitioning()
         $rows = iterator_to_array($qr->rows());
         $this->assertEquals(50, $rows[0]['ct']);
     }
+
+    /**
+     * @depends testInsertRowToTable
+     */
+    public function testJsonDatatype()
+    {
+        $rows = iterator_to_array(self::$table->rows());
+        $actualRow = $rows[self::$expectedRows-1];
+
+        $this->assertEquals(
+            json_decode($this->row['Json'], true)['HouseNumber'],
+            json_decode($actualRow['Json']->get(), true)['HouseNumber']
+        );
+        $this->assertEquals(
+            json_decode($this->row['Json'], true)['City'],
+            json_decode($actualRow['Json']->get(), true)['City']
+        );
+    }
+
+    public function testInsertFaultyJsonToTable()
+    {
+        $row = $this->row;
+        $row['Json'] = '{';
+        $insertResponse = self::$table->insertRow($row);
+        $this->assertEquals(
+            $insertResponse->info()['insertErrors'][0]['errors'][0]['reason'],
+            "invalid"
```

Comment:
Is there something more to assert?
Like invalid json, etc rather than just `invalid` word.

Reply:
I can assert the the message also then..

Reply:
Added